packages/open-payments/README.md

```diff
@@ -0,0 +1,249 @@
+| `keyId`             | The key identifier of the given private key and the corresponding public JWK document.                                                                                                                                                                                                                                                                                                                                                                                   |
+
+> **Note**
+>
+> To simplify EdDSA-Ed25519 key provisioning and JWK generation, you can use the `parseOrProvisionKey` and `generateJwk` methods from the [`http-signature-utils`](../http-signature-utils/README.md#usage) package.
+
+## Example
+
+As mentioned previously, Open Payments APIs can facilitate a payment between two parties.
+
+For example, say Alice wants to purchase a $50 product from a merchant called Shoe Shop. If both parties have Open Payments enabled wallets, where Alice's payment pointer is `https://cloud-nine-wallet/alice`, and Shoe Shop's is `https://happy-life-bank/shoe-shop`, requests during checkout from Shoe Shop's backend would look like this with the client:
+
+1. Create an Open Payments client
+
+In this case, since Shoe Shop wants to make requests that require authorization, Shoe Shop's backend will need to create an `AuthenticatedClient`:
+
+```ts
+import { createAuthenticatedClient } from '@interledger/open-payments'
+
+const client = await createAuthenticatedClient({
+  paymentPointerUrl: 'https://happy-life-bank/shoe-shop',
+  keyId: KEY_ID,
+  privateKey: PRIVATE_KEY
+  // The public JWK with this key (and keyId) would be available at https://happy-life-bank/shoe-shop/jwks.json
+})
+```
+
+2.  Get `PaymentPointers`
+
+Grab the payment pointers of the parties:
+
+```ts
+const shoeShopPaymentPointer = await client.paymentPointer.get({
+  url: 'https://happy-life-bank/shoe-shop'
+})
+
+const customerPaymentPointer = await client.paymentPointer.get({
+  url: 'https://cloud-nine-wallet/alice'
+})
+```
+
+3. Create `IncomingPayment`
+
+Shoe Shop's backend gets a grant (if it doesn't have one already) to create an `IncomingPayment` on its wallet:
+
+```ts
+const incomingPaymentGrant = await client.grant.request(
+  { url: shoeShopPaymentPointer.authServer },
+  {
+    access_token: {
+      access: [
+        {
+          type: 'incoming-payment',
+          actions: ['read-all', 'create']
+        }
+      ]
+    }
+  }
+)
+
+const incomingPayment = await client.incomingPayment.create(
+  {
+    paymentPointer: shoeShopPaymentPointer.authServer,
+    accessToken: incomingPaymentGrant.access_token.value
+  },
+  {
+    incomingAmount: {
+      assetCode: 'USD',
+      assetScale: 2,
+      value: '5000'
+    },
+    description: 'Purchase at Shoe Shop',
+    externalRef: '#INV2022-8363828'
+  }
+)
+```
```

Comment:
Should the example use a third-party client?
As is, I'm wondering why Shoe Shop has to use Open Payments to create an incoming payment for itself.

Reply:
I'm curious if a bank would customers access other than via Open Payments.
For instance to create incoming payments

Reply:
I think Amazon Marketplace is a good example here - I'll convert to that. I was thinking like Shopify at first, but in that example, the merchant shop/wallet boundary gets blurred again.

Reply:
I wonder if it'd be safer to still use a fictitious name to avoid any issues around implications that FAANGs are (already) using our technology